src/problems/shinny_sam_sudoku.clj

```diff
@@ -0,0 +1,96 @@
+(ns problems.shinny-sam-sudoku)
+
+(defrecord State [board curr-position])
+
+;check row/column for duplicates
+(defn duplicates? [coll]
+  (not (apply distinct? (filter #(> % 0) coll))))
+
+
+;return a row
+(defn row-seq [board y]
+  (nth board y))
+
+;return a column
+(defn col-seq [board x]
+  (nth (apply map vector  board) x))
+
+;check if a row, column or 3x3 is legal
+(defn legal? [coll]
+  (and (not (duplicates? coll))))
+
+
+;put numbers into the current position of the board
+(defn add [board position number]
+    (assoc-in board position number))
+
+
+;update the current position of the board
+(defn updatePos [state]
+  (let [x (first (:curr-position state))
+        y (second (:curr-position state))]
+    (if (= 0 (compare (:curr-position state) [8, 8]))
+      [8, 8]
+      (if (= y 8)
+      [(+ x 1) (- y 8)]
+      [x (+ y 1)]))))
+
+;divide into 3*3 and return a coll
+(defn grid [board hpos vpos]
+  (for [x[(* hpos 3) (+ (* hpos 3) 1) (+ (* hpos 3) 2)] y[(* vpos 3) (+ (* vpos 3) 1) (+ (* vpos 3) 2)]]
+    (get-in board [x y])))
+
+
+;verify if it is legal and return a coll of true or false
+(defn grid-legal? [state]
+  (for [x[0 1 2] y[0 1 2]]
+    (legal? (grid (:board state) x y))))
+
+;check if the whole board is legal
+(defn board-legal? [state]
+  (let [row (row-seq (:board state) (first (:curr-position state)))
+        col (col-seq (:board state) (second (:curr-position state)))]
+            (and (legal? row) (legal? col) (every? true? (grid-legal? state)))))
+
+;children function
+(defn children [state]
+  (let [x (first (:curr-position state))
+        y (second (:curr-position state))]
+    (for [num [1 2 3 4 5 6 7 8 9]
+          :let [new-state (add (:board state) (:curr-position state) num)]
+          :when (board-legal? (->State new-state [x y]))]
+      (if (not= 0 (get-in (:board state) [x y])) (->State (:board state) (updatePos state))
+      (->State new-state (updatePos state))))))
+
+
+;heuristic helper function
```

Comment:
This comment really isn't helpful, and nor is the comment on line 73. Maybe a comment telling us a little bit about what your heuristic function _does_ or why you chose it would have been more useful?